tlv/record_type.go

```diff
@@ -0,0 +1,59 @@
+package tlv
+
+import (
+	"github.com/btcsuite/btcd/btcec/v2"
+	"golang.org/x/exp/constraints"
+)
+
+// Primitive is a type constraint that capture the set of "primitive" types,
+// which are the built in stdlib types, and type defs of those types
+type Primitive interface {
+	constraints.Integer | ~[]byte | ~[32]byte | ~[33]byte | ~bool |
```

Comment:
`constraints.Integer` includes both Signed and Unsigned integers. Since `MakePrimitiveRecord` only currently supports unsigned integers, I think we should just do `constraints.Unsigned` here. Cause otherwise if we use `int16` for example, we will get a panic at runtime (`panic: unknown primitive type: *int16`) instead of a compile time error